prow/kube/types.go

```diff
@@ -109,6 +109,8 @@ type Volume struct {
 	DownwardAPI *DownwardAPISource `json:"downwardAPI,omitempty"`
 	HostPath    *HostPathSource    `json:"hostPath,omitempty"`
 	ConfigMap   *ConfigMapSource   `json:"configMap,omitempty"`
+	// https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/types.go#L954
+	DefaultMode int32 `json:"defaultMode,omitempty"`
```

Comment:
this is in the wrong object - `defaultMode` is part of `SecretVolumeSource` (`SecretSource` here), not `Volume`.

(this illustrates the need for the kubernetes client library. :)